pkg/clients/imagebuilder/client.go

```diff
@@ -410,3 +427,38 @@ func (c *Client) GetMetadata(image *models.Image) (*models.Image, error) {
 	c.log.Infof("Done with metadata for image")
 	return image, nil
 }
+
+// GetThirdPartyRepos finds the url of Third Party Repository using the name
+func (c *Client) GetThirdPartyRepos(image *models.Image) ([]Repository, error) {
+	account, err := common.GetAccountFromContext(c.ctx)
+	if err != nil {
+		c.log.WithField("error", err).Error("error retrieving account")
+		return nil, errors.New("error retrieving account")
+	}
+
+	repos := make([]Repository, len(image.ThirdPartyRepositories))
+	thirdpartyrepos := make([]models.ThirdPartyRepo, len(image.ThirdPartyRepositories))
+	thirdpartyrepoIDS := make([]int, len(image.ThirdPartyRepositories))
+
+	for repo := range image.ThirdPartyRepositories {
+		thirdpartyrepoIDS[repo] = int(image.ThirdPartyRepositories[repo].ID)
+	}
+	var count int64
+	result := db.DB.Where("account = ?", account).Find(&thirdpartyrepos, thirdpartyrepoIDS).Count(&count)
+	if result.Error != nil {
+		log.Error(result.Error)
+		return nil, result.Error
+	}
+
+	if count != int64(len(thirdpartyrepoIDS)) {
+		return nil, errors.New("enter valid third party repository id")
+	}
+	for repoID := 0; repoID < len(thirdpartyrepos); repoID++ {
```

Comment:
This is not repoID just ind 
```suggestion
	for int := 0; ind < len(thirdpartyrepos); ind++ {
```